### PR TITLE
Add Zambia MDA Reports

### PR DESCRIPTION
**Pull request by moshthepitt**
This PR adds support for Zambia Dynamic-MDA (aka NTD) reports as defined in [this doc](https://docs.google.com/presentation/d/1GlKZzz65XsGsiWNP531x_XK99L1Zg8VNM8-uJgj_29k/edit#slide=id.g8f6a6020ac_0_52).

### New env vars

```
REACT_APP_SUPERSET_DYNAMIC_MDA_REPORTING_PLANS_SLICE=596
REACT_APP_SUPERSET_DYNAMIC_MDA_REPORTING_JURISDICTIONS_DATA_SLICES=593
REACT_APP_SUPERSET_DYNAMIC_MDA_REPORTING_JURISDICTIONS_COLUMNS=zambiaMDAUpper2020
REACT_APP_SUPERSET_DYNAMIC_MDA_REPORTING_FOCUS_AREAS_COLUMNS=zambiaMDALower2020
REACT_APP_SUPERSET_DYNAMIC_MDA_REPORTING_JURISDICTIONS_FOCUS_AREA_LEVEL=3
```

### Linked issues

Fix: #1110 Fix: #1111  Fix: #1112 Fix: #1113

### CODE REVIEW

**Comment by moshthepitt** [comment]
## Screenshots

### Plans list 

mda plans

### Higher level jurisdictions

mda higher level

### Lower level jurisdictions

mda lower level

File: src/App/tests/App.test.tsx [hunk 1]
```diff
@@ -24,25 +25,27 @@ describe('App', () => {
     history.push('/');
   });
 
+  const LogoutComponent = () => null;
```
Comment (moshthepitt) [inline src/App/tests/App.test.tsx:28]: Added this to ensure no linter warnings.

File: src/components/TableCells/JurisdictionReport.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { DropDownCellProps } from '@onaio/drill-down-table';
```
Comment (moshthepitt) [inline src/components/TableCells/JurisdictionReport.tsx:1]: Intention is for this component to be used whenever we don't need anything fancy: either just a link or plain text

File: src/helpers/indicators.tsx [hunk 3]
```diff
@@ -124,31 +124,6 @@ export function getFIAdherenceIndicator(cell: Cell) {
...
-      className="137-container"
-      style={{
-        color: cell.value > ZERO ? GREEN : cell.value < ZERO ? RED : ORANGE,
-      }}
-    >
-      {cell.value}d to go
-    </div>
-  );
-}
-
```
Comment (moshthepitt) [inline src/helpers/indicators.tsx:151]: This is dead code thus removing it.

File: src/helpers/tests/indicators.test.tsx [hunk 4]
```diff
@@ -1,13 +1,14 @@
 import { map } from 'lodash';
 import { PERSONS, STRUCTURES } from '../../configs/lang';
+import { Goal } from '../../store/ducks/goals';
```
Comment (moshthepitt) [inline src/helpers/tests/indicators.test.tsx:3]: Fix linter warnings

File: src/store/ducks/generic/MDAPlans.ts [hunk 5]
```diff
@@ -0,0 +1,106 @@
+import {
```
Comment (moshthepitt) [inline src/store/ducks/generic/MDAPlans.ts:1]: This duck is very similar to the one for IRS and MDA-Point.  We can make this more DRY: https://github.com/onaio/reveal-frontend/issues/1143

File: src/containers/pages/GenericJurisdictionReport/index.tsx [hunk 6]
```diff
@@ -200,7 +201,7 @@ const GenericJurisdictionReport = (
 
   let columnsToUse = get(plansTableColumns, jurisdictionColumn, null);
   if (currLevelData && currLevelData.length > 0) {
-    if (currLevelData[0].jurisdiction_depth === +focusAreaLevel) {
+    if (currLevelData[0].jurisdiction_depth >= +focusAreaLevel) {
```
Comment (moshthepitt) [inline src/containers/pages/GenericJurisdictionReport/index.tsx:204]: This works for now, but we need to make it better.  See: https://github.com/onaio/reveal-frontend/issues/1144

File: src/configs/env.ts [hunk 7]
```diff
@@ -57,6 +57,10 @@ export type ENABLE_TEAMS = typeof ENABLE_TEAMS;
 export const ENABLE_MDA_POINT = process.env.REACT_APP_ENABLE_MDA_POINT === 'true';
 export type ENABLE_MDA_POINT = typeof ENABLE_MDA_POINT;
 
+/** Do you want to enable the MDA Point plan features? */
```
Comment (ciremusyoka) [inline src/configs/env.ts:60]: Please update this docstring

File: src/configs/lang.ts [hunk 8]
```diff
@@ -37,6 +37,7 @@ export const HOME_TITLE = translate('HOME_TITLE', 'Home page');
 export const IRS_TITLE = translate('IRS_TITLE', 'IRS');
 export const PLANNING_PAGE_TITLE = translate('PLANNING_PAGE_TITLE', 'Planning tool');
 export const IRS_REPORTING_TITLE = translate('IRS_REPORTING_TITLE', 'IRS Reporting');
+export const MDA_REPORTING_TITLE = translate('MDA_REPORTING_TITLE', 'MDA Reporting');
```
Comment (ciremusyoka) [inline src/configs/lang.ts:40]: Lets add this to the `en.json` file too.

File: src/containers/pages/DynamicMDA/plans/index.tsx [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+import { GenericPlan } from '../../../../store/ducks/generic/plans';
+import { GenericPlanListProps, GenericPlansList } from '../../GenericPlansList';
+
+/** register the MDA plan definitions reducer */
+reducerRegistry.register(DynamicMDAPLansReducerName, DynamicMDAPLansReducer);
+
+/** a list of plan statuses to be displayed */
+const allowedPlanStatusList = getPlanStatusToDisplay(HIDDEN_PLAN_STATUSES);
+
+/** Simple component that loads a preview list of IRS plans */
```
Comment (ciremusyoka) [inline src/containers/pages/DynamicMDA/plans/index.tsx:28]: Update this

File: src/components/page/Header/index.tsx [hunk 10]
```diff
@@ -216,6 +219,19 @@ export class HeaderComponent extends React.Component<HeaderProps, State> {
                         </DropdownItem>
                       </div>
                     )}
+                    {ENABLE_DYNAMIC_MDA && (
```
Comment (p-netm) [inline src/components/page/Header/index.tsx:222]: should we also include this in the check of when the `monitor`-tab dropdown is rendered in the header navigation?

File: src/store/ducks/generic/tests/MDAPlans.test.ts [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+      plan_title: 'BERG',
+    };
+
+    expect(getPlansArrayByTitle()(store.getState(), titleFilter)).toEqual([
+      fixtures.DynamicMDAPlans[2],
+    ]);
+    expect(getPlansArrayByTitle()(store.getState(), titleUpperFilter)).toEqual([
+      fixtures.DynamicMDAPlans[2],
+    ]);
+    expect(plansArraySelector(store.getState(), { plan_title: 'Berg' })).toEqual([
```
Comment (p-netm) [inline src/store/ducks/generic/tests/MDAPlans.test.ts:57]: seems like this assertion is repeated L51

**Comment by moshthepitt** [comment]
@p-netm @ciremusyoka I addressed your comments.  Please re-review

**Pull request merged**